src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs

```diff
@@ -105,7 +107,20 @@ public string GetId()
                 if (_addedFields != _countOfFields)
                     ThrowNumberOfProcessedFieldsMismatch();
 
-                return _id.AppendFormat(_formattedPattern, _values).ToString();
+                var id = _id.AppendFormat(_formattedPattern, _values).ToString();
+                
+                ValidateId(id);
+
+                return id;
+            }
+
+            public static void ValidateId(string id)
+            {
+                if (id.EndsWith('/'))
```

Comment:
After merging this to 5.0 we need to use DocumentDatabase.IndentityPartsSeparator here

Reply:
Okay. Added TODO in issue comment